### DISCUSSION TITLE
The structure behind development

### DISCUSSION THREAD

**Discussion by davehakkens** [ONEARMY/community-platform#1109] [General]
The contributions you guys bring to the platform are very valuable. We see a lot of potential in the community platform and it will be a crucial piece of infrastructure for our future projects from One Army. (For us the Precious Plastic community is just the first prototype) That said, we come from a maker background. We are used to working in workspaces, making machines, in real life with materials. Our team has many makers, engineers, designers etc. Coding is a skillset we are not very familiar with. And on top of this it generally means more remote work and contributions. We love it, see a lot of potential in it and see the Open Source software systems as big inspiration for our hardware. 

Long story short. We at One Army are beginners in web development.
We need to organise ourselves, grow a steady team and make releases more regular.

**💰 The bounty system.**
We started using (internally) a bounty system, to pay contributers for the work that is done.  We set out specific tasks and issues and rewards developers that take them on. This works quite well, but currently, we manage ourselves in a spreadsheet. Not very smooth or transparent. We are intrigued by [Open Collective,](https://opencollective.com/) it seems like it could structure us more efficiently. But we never worked with it. Do you guys have any suggestions on this? @paulpichot @tudi2d @danitrod 

**⚙️ Maintainers**
This is a big bottleneck for us. When external people push commits or want to contribute they often need help. Guidelines, reviews and tips for improvements. @BenGamma  did this for a while but currently @chrismclarke is taking on this task. However this is a lot for him to handle especially with a growing amount of developers contributing. He works in the backend to make this process smoother, setting up live builds and previews. But in general, we could use more long-term maintainers for the platform. People that understand the code, libraries and structure. A complex task, we haven’t got a solution for this yet, just throwing it here so you are aware of the issue :)

**💻 Monthly calls.**
We started doing monthly dev call. Every first monday of the month at 6PM GMT. We list out problems, new versions and just to catch up with the team beyond a keyboard. Well still kinda on a keyboard but we add a screen.  If you are interested to join let me know below.

This topic is mainly to share where we are currently at and the direction we are thinking about. But always up for suggestions from some real devs ;) 

For those that want to see what we want to develop next. Go to this topic.
[The next big developments for the platform](https://github.com/ONEARMY/community-platform/discussions/1108)

**Comment by davehakkens** [comment]
Also is there a way to reach you @danitrod? Got some open bounty money for you 🎁

**Reply by danitrod** [reply]
@davehakkens neat! Feel free to reach me at danitrod@protonmail.com. About the bounty system idea, unfortunately I have no experience with, so I'm probably not the best person to help. I am relatively new to open source, this was the first big repository to which I contributed, and I was very happy to help you guys. I look forward to continue contributing as long as I'm bringing any value, and I'd be happy to join some of those monthly dev calls.
Also, wanted to give a big thanks to you and @chrismclarke for the warm welcoming you gave me here and for your dedication to build this awesome platform. Hopefully I can continue to help you guys along.

**Reply by chrismclarke** [reply]
Thanks for all the contributions!

**Comment by chrismclarke** [comment]
Thanks for capturing all this @davehakkens, 

Also would be good to have more maintainers so that any code I write can also get another look at in case I start going rogue (:D). I think the most important thing for a maintainer is less a deep technical knowledge (if there's something in the code the maintainer doesn't understand then likely others won't either, and so the dev should probably add better comments to make it clearer for everyone), more just a decent feel for "does the update do what it says it is doing" and "does the code make logical sense". 

So if anyone is interested in doing more of this but feeling uncertain I can easily start with assigning reviews for specific PRs that should hopefully be a bit more straight-forward (smaller frontend changes etc.). Right now I think my overall time on reviews is maybe around 2h per week, so even small additions most welcome.

**Comment by davehakkens** [comment]
Just a little update on this. We did start using Open Collective :)
https://opencollective.com/onearmy
Let's see

**Comment by always-tinkering** [comment]
Hey all, can I get an invite to the dev meeting tomorrow? 

I am out of development practice, although have been spinning back up on basic web development.Currently studying Node/React/TS to give some assistance on the Platform. Although, I do feel the role of maintainer would work very well. I will continue to hone my dev skills again, but in the meantime I can review PRs for their features/bug fixes.

**Reply by chrismclarke** [reply]
@TinkeringAlways 
Hi, only just seeing this, apologies. You'd definitely be most welcome to join us if you happen to see this in time. Really need to find somewhere better to share this info!

https://meet.google.com/tky-eegn-stw

**Reply by davehakkens** [reply]
start in 10 min https://meet.google.com/tky-eegn-stw

**Reply by davehakkens** [reply]
haha good timing @chrismclarke

**Reply by FrannyMendesLevitin** [reply]
Hey Dev team! Are you all still open to inviting people to the dev team meetings? I'm most experienced with front-end, but I'd be excited to support however it's useful.

**Reply by chrismclarke** [reply]
Hi @FrannyMendesLevitin, you'd be totally welcome to join. The next call will be on Monday October 4th at 6pm Portugal time (same as UK), hosted at the same link as above (https://meet.google.com/tky-eegn-stw?hs=224). 

We also should have various issues tagged `help wanted` and `good first issue` if you wanted to start looking into the main codebase and dev priorities. If you have any issues you can either try tag me in a comment on here (although I probably don't check as often as I would like), or ask for an invite to slack and we have a channel on there which is more active.
